### DISCUSSION TITLE
When should the Sender header and From header be set?

### DISCUSSION THREAD

**Discussion by kikaragyozov** [jstedfast/MailKit#1828] [Q&A] [answered]
I have an SDK wrapper on top of MailKit that handles SMTP-related things, like authentication, sending of emails etc.

1. If authentication is required, and Username and Password are specified, and the Username is an e-mail address:
   - Use the Username as the value for the `From` header.

2. If authentication is required, Username is an e-mail address, but the client has explicitly specified a value for the `From` header:
   - Set the `Sender` header to the Username.
   - Set the `From` header to the client's specified value.

3. If authentication is required, but the Username is not an e-mail address, and the client has not specified a `From` header:
   - Throw an exception.

4. If the client has specified a value for the `From` header:
   - Set the `From` header to the specified value.

Are all of these states valid and sound in reasoning? Or is something flawed here?

**Comment by jstedfast** [comment, marked as answer]
1. Sounds reasonable.
2. I would recommend only setting the `Sender` if the Username value differs from the `From` value specified by the user.
3. Yep.
4. Yep.

To elaborate a bit, the `Sender` header should only ever be used if it differs from the `From` header.
